sdk/src/main/java/io/opentdf/platform/sdk/ZipReader.java

```diff
@@ -184,8 +193,8 @@ public int read() throws IOException {
                         return -1;
                     }
                     setChannelPosition();
-                    while (buf.position() != buf.capacity()) {
-                        if (zipChannel.read(buf) < 0) {
+                    while (buf.hasRemaining()) {
+                        if (zipChannel.read(buf) <= 0) {
```

Comment:
This probably isn't a big deal in the cases we are dealing with here but according to the docs for `read` the channel can return 0 bytes before it's been exhausted.

Reply:
Thank you for calling this out, this is a discreet CPU usage DoS concern.  You're correct that a zero could be returned for non-blocking streams.  Blocking implementations are guaranteed to block until at least one byte is available, or the end of the stream is reached.

The non-blocking behavior where a zero is potentially returned, results in a tight loop here.  Right now this implementation is not well designed for non-blocking implementations.  And since a tight loop is never ideal, I feel like this change to only support blocking streams is the safest and likely least impactful.

Let me know if you have any thoughts on how this can be handled better.  Thank you!

Reply:
Yeah, I think it make sense to trade off correctness for susceptibility to DoS here. In either case it seems like a lot of work to create something that is both seekable and blocks outside of NFS or something so this doesn't seem to likely.